File: backend/app/schemas/vendor.py
"""Pydantic schemas for vendor management."""
from pydantic import BaseModel, ConfigDict, Field
from datetime import datetime
from uuid import UUID
from typing import Optional, List


class VendorCreate(BaseModel):
    """Schema for creating a vendor."""
    vendor_code: Optional[str] = Field(None, min_length=1, max_length=50)
    company_name: str = Field(..., min_length=1, max_length=200)
    country: str = Field(..., min_length=1, max_length=100)
    certifications: Optional[List[str]] = None
    product_categories: Optional[List[str]] = None
    credibility_score: Optional[int] = Field(50, ge=0, le=100)
    on_time_delivery_rate: Optional[float] = Field(None, ge=0.0, le=1.0)
    quality_score: Optional[int] = Field(None, ge=0, le=100)
    avg_lead_time_days: Optional[int] = None
    primary_contact_name: Optional[str] = None
    primary_contact_email: Optional[str] = None
    primary_contact_phone: Optional[str] = None
    payment_terms: Optional[str] = None
    notes: Optional[str] = None


class VendorUpdate(BaseModel):
    """Schema for updating a vendor."""
    vendor_code: Optional[str] = Field(None, min_length=1, max_length=50)
    company_name: Optional[str] = Field(None, min_length=1, max_length=200)
    country: Optional[str] = Field(None, min_length=1, max_length=100)
    certifications: Optional[List[str]] = None
    product_categories: Optional[List[str]] = None
    credibility_score: Optional[int] = Field(None, ge=0, le=100)
    on_time_delivery_rate: Optional[float] = Field(None, ge=0.0, le=1.0)
    quality_score: Optional[int] = Field(None, ge=0, le=100)
    avg_lead_time_days: Optional[int] = None
    primary_contact_name: Optional[str] = None
    primary_contact_email: Optional[str] = None
    primary_contact_phone: Optional[str] = None
    payment_terms: Optional[str] = None
    is_active: Optional[bool] = None
    notes: Optional[str] = None


class VendorResponse(BaseModel):
    """Schema for vendor response."""
    model_config = ConfigDict(from_attributes=True)

    id: UUID
    company_id: UUID
    vendor_code: str
    company_name: str
    country: str
    certifications: Optional[List[str]]
    product_categories: Optional[List[str]]
    credibility_score: int
    on_time_delivery_rate: Optional[float]
    quality_score: Optional[int]
    avg_lead_time_days: Optional[int]
    primary_contact_name: Optional[str]
    primary_contact_email: Optional[str]
    primary_contact_phone: Optional[str]
    payment_terms: Optional[str]
    is_active: bool
    notes: Optional[str]
    created_at: datetime
    updated_at: datetime


class VendorListResponse(BaseModel):
    """Schema for vendor list response."""
    total: int
    items: List[VendorResponse]

File: frontend/lib/validations/vendor.ts
/**
 * Zod validation schemas for vendor forms
 */

import * as z from "zod"

export const vendorFormSchema = z.object({
  vendor_code: z.string().optional(),
  company_name: z.string().min(1, "Company name is required"),
  country: z.string().min(1, "Country is required"),
  certifications: z.array(z.string()).optional(),
  product_categories: z.array(z.string()).optional(),
  credibility_score: z.number().min(0).max(100).optional(),
  on_time_delivery_rate: z.number().min(0).max(100).optional(),
  quality_score: z.number().min(0).max(100).optional(),
  avg_lead_time_days: z.number().min(0).optional(),
  primary_contact_name: z.string().optional(),
  primary_contact_email: z.string().email().optional().or(z.literal("")),
  primary_contact_phone: z.string().optional(),
  payment_terms: z.string().optional(),
  is_active: z.boolean().optional(),
  notes: z.string().optional(),
})

export type VendorFormValues = z.infer<typeof vendorFormSchema>
